src/app/pages/send/send-crypto-asset-form/send-form.utils.ts

```diff
@@ -1,3 +1,44 @@
+import BigNumber from 'bignumber.js';
+
+import { Money } from '@shared/models/money.model';
+
+import { convertAmountToBaseUnit } from '@app/common/money/calculate-money';
+
 export function createDefaultInitialFormValues<T extends Record<string, unknown>>(obj: T) {
   return { amount: '', recipient: '', ...obj };
 }
+
+interface GetSendAllAmountArgs {
+  balance: Money;
+  fee: number | string;
+  fontSize: number;
+  fontSizeModifier: number;
+  maxFontSize: number;
+  pendingTxsBalance: BigNumber;
+}
+export function getSendAllAmount(props: GetSendAllAmountArgs) {
+  const { balance, fee, fontSize, fontSizeModifier, maxFontSize, pendingTxsBalance } = props;
+  let convertedAmount = '';
+
+  switch (balance.symbol) {
+    case 'BTC':
+      convertedAmount = convertAmountToBaseUnit(balance).minus(fee).toString();
+      break;
+    case 'STX':
+      convertedAmount = convertAmountToBaseUnit(balance)
+        .minus(pendingTxsBalance)
+        .minus(fee)
+        .toString();
+      break;
+    default:
+      convertedAmount = convertAmountToBaseUnit(balance).toString();
+      break;
+  }
```

Comment:
How can we do this without a switch?

Reply:
Yep, I keep using switch statements thinking it is the best way to scale the code if we need to with other blockchains later.

Reply:
I get your point, but I did have to remove the `rightInputOverlay` you added to pass in the `SendAllButton` bc the send all button has to trigger resizing the font in the `AmountField`. It made more sense to define `onSendAll` in the amount field component. Also, the fee is only known inside the form and not in the parent to pass in already calculated. I am not sure how to know inside the amount field if I should subtract the fee or not, if I don't use a conditional of some kind? Let me know if you see something I am missing here.

Reply:
Nvm, after I wrote that I think I see a way I overlooked.